cms/datasets/views.py

```diff
@@ -0,0 +1,109 @@
+from collections.abc import Iterable
+from typing import Any
+
+from django import forms
+from django.db.models import QuerySet
+from django.views import View
+from wagtail.admin.forms.choosers import BaseFilterForm
+from wagtail.admin.ui.tables import Column
+from wagtail.admin.views.generic.chooser import (
+    BaseChooseView,
+    ChooseResultsViewMixin,
+    ChooseViewMixin,
+    ChosenResponseMixin,
+    ChosenViewMixin,
+    CreationFormMixin,
+)
+from wagtail.admin.viewsets.chooser import ChooserViewSet
+
+from cms.datasets.models import Dataset, ONSDataset
+
+
+class DatasetBaseChooseViewMixin:
+    @property
+    def columns(self):  # type: ignore
+        return [
+            *getattr(super(), "columns", []),
+            Column("edition", label="Edition", accessor="formatted_edition"),
+            Column("version", label="Latest Version", accessor="version"),
+        ]
+
+
+class DatasetSearchFilterForm(BaseFilterForm):
+    q = forms.CharField(
+        label="Search datasets",
+        widget=forms.TextInput(attrs={"placeholder": "Dataset title"}),
+        required=False,
+    )
+
+    def filter(self, objects: Iterable[Any]) -> Iterable[Any]:
+        objects = super().filter(objects)
+        search_query = self.cleaned_data.get("q")
+        if search_query:
+            search_query_lower = search_query.strip().lower()
+            objects = [obj for obj in objects if self.obj_matches_search_query(obj, search_query_lower)]
+            self.is_searching = True
+            self.search_query = search_query
+        return objects
```

Comment:
Worth adding a comment to say, this is done in code as the API doesn't support filtering, but will be supported by the API in the future.